### ISSUE TITLE
Windows linux subsystem support

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle-native#298]
This is a place to collect details of what it would mean to support the WLS.

From a modelling perspective, the WLS is a _linux_ host that also happens to be able to run binaries that _target_ windows. It's not a windows host. This means that the native plugins should build, test and publish linux binaries using the linux tooling and system APIs. Which is currently how they behave on the WLS so "support" could just amount to some CI test coverage and documentations. Certainly a good place to start.

The native plugins should also be able to cross-compile to build, test and publish windows binaries using either the windows tooling (visual c++, mingw, clang, clang/gcc on cygwin) or linux tooling (mingw hosted on linux) and the windows system APIs.

The native plugins should also be able to generate visual studio configuration from the WLS.

In other words, the native plugins should treat the WLS as linux as far as defaults go, and offer every capability that is available on linux _and_ every capability that is available on windows.
